django/utils/datastructures.py

```diff
@@ -286,7 +286,7 @@ def _destruct_iterable_mapping_values(data):
             )
         if not isinstance(elem[0], str):
             raise ValueError('Element key %r invalid, only strings are allowed' % elem[0])
-        yield tuple(elem)
+        yield elem if isinstance(elem, Iterable) else tuple(elem)
```

Comment:
No tests fail if I remove this change. I presume it was to avoid calling `tuple()` unnecessarily?

For that matter, we can just change to the following as we're always unpacking into key-value pairs:

```suggestion
        yield elem
```

I don't think we need any guarantees that this will return a `tuple`. Loosely it only needs return an iterable of two elements with the first being a string.

Reply:
Thanks. I dropped calling `tuple`